iceoryx_utils/test/moduletests/test_cxx_deadline_timer.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/cxx/deadline_timer.hpp"
+#include "iceoryx_utils/internal/units/duration.hpp"
+#include "test.hpp"
+#include "testutils/timing_test.hpp"
+
+#include <chrono>
+#include <thread>
+
+using namespace ::testing;
+
+using namespace iox::units;
+using namespace iox::units::duration_literals;
+
+using Timer = iox::cxx::DeadlineTimer;
+
+class DeadlineTimer_test : public Test
+{
+  public:
+    virtual void SetUp()
+    {
+        numberOfCalls = 0;
+    }
+
+    virtual void TearDown()
+    {
+    }
+
+    Duration second{1_s};
+
+    std::atomic<int> numberOfCalls{0};
+    static const Duration TIMEOUT;
+    static const int SLEEPTIME;
+};
+
+const Duration DeadlineTimer_test::TIMEOUT{10_ms};
+const int DeadlineTimer_test::SLEEPTIME = DeadlineTimer_test::TIMEOUT.milliSeconds<int>();
+
+TIMING_TEST_F(DeadlineTimer_test, DurationOfNonZeroIsExpiresAfterTimeout, Repeat(5), [&] {
```

Comment:
What if the Deadline timer has a timeout of zero? Is the Callback then called as fast as possible? We should also always keep the following test ideas in mind:
- Zero
- One
- Many
- Corner Cases
- Limits

to cover every aspect of a class.

Reply:
This is only the deadline timer use case. in case if the timer has a duration set to zero the end time will be the current time and the timer will be already expired. I can add a test case for this.